dnscache.go

```diff
@@ -6,22 +6,27 @@ import (
 	"sync"
 	"time"
 
-	"github.com/rs/dnscache/internal/singleflight"
+	"golang.org/x/sync/singleflight"
```

Comment:
I'd prefer not to add an external dependency for this.

Reply:
Why do you don't want the official `golang.org/x/sync` dependency? In my opinion, the great benefit is a maintained and an official dependency.

Reply:
Well, I guess you're right. I tend to avoid small dependencies like that, but in this case I guess it make sense.